source/common/config/utility.cc

```diff
@@ -159,35 +160,8 @@ std::string Utility::resourceName(const ProtobufWkt::Any& resource) {
       fmt::format("Unknown type URL {} in DiscoveryResponse", resource.type_url()));
 }
 
-std::vector<Stats::TagExtractorPtr>
-Utility::createTagExtractors(const envoy::api::v2::Bootstrap& bootstrap) {
-  std::vector<Stats::TagExtractorPtr> tag_extractors;
-
-  // Ensure no tag names are repeated.
-  std::unordered_set<std::string> names;
-  auto add_tag = [&names, &tag_extractors](const std::string& name, const std::string& regex) {
-    if (!names.emplace(name).second) {
-      throw EnvoyException(fmt::format("Tag name '{}' specified twice.", name));
-    }
-
-    tag_extractors.emplace_back(Stats::TagExtractorImpl::createTagExtractor(name, regex));
-  };
-
-  // Add defaults.
-  if (!bootstrap.stats_config().has_use_all_default_tags() ||
-      bootstrap.stats_config().use_all_default_tags().value()) {
-    for (const std::pair<std::string, std::string>& default_tag :
-         TagNames::get().name_regex_pairs_) {
-      add_tag(default_tag.first, default_tag.second);
-    }
-  }
-
-  // Add custom tags.
-  for (const envoy::api::v2::TagSpecifier& tag_specifier : bootstrap.stats_config().stats_tags()) {
-    add_tag(tag_specifier.tag_name(), tag_specifier.regex());
-  }
-
-  return tag_extractors;
+Stats::TagProducerPtr Utility::createTagProducer(const envoy::api::v2::Bootstrap& bootstrap) {
+  return Stats::TagProducerPtr{new Stats::TagProducerImpl(bootstrap.stats_config())};
```

Comment:
nit: std::make_unique

Reply:
I see. Always prefer std::make_unique? Maybe same here https://github.com/envoyproxy/envoy/pull/2357/files#diff-4112b28a18267da2a53245cfbacaeb94R16.

Reply:
We didn't have make_unique until we switched to c++14, so there are tons of places where it could be used. I wouldn't go out of your way, but in new cases I would use it.